metadata/src/lib.rs

```diff
@@ -93,6 +93,14 @@ pub struct Album {
     pub covers: Vec<FileId>,
 }
 
+#[derive(Debug, Clone)]
+pub struct Playlist {
+    pub user: String,
+    pub length: i32,
```

Comment:
What is `length` here? Is it the number of tracks? Wouldn't we normally omit that and have users do `tracks.len()` instead? Or is it something else?

Reply:
Yes it's just the number of tracks. I included it just because it's provided separately in the protobuf message for some reason. It's pretty redundant so I'll remove it.

Reply:
I was just thinking, is there any paging mechanism for very long playlists where the tracks returned in `tracks` is actually only the first x tracks with the total length being returned in that `length` field? This is what they do for their web API and also for the spirc queue.

Reply:
Yes the web api is limited to I think 100 tracks per request. I don't think there is anything like that here, but perhaps I should test it on a really long playlist to make sure it works.

Reply:
Just tested on a 701 track playlist. I got all tracks right away, so I can probably safely remove length.

Reply:
Maybe have a check to see that we've received all the tracks and log a warning if not, in case this ever changes.

Reply:
Very good! I'll do that.